src/edit_session.js

```diff
@@ -2366,6 +2366,20 @@ class EditSession {
         };
     }
 
+    /**
+     * @returns {string} the last character preceding the cursor in the editor
+     */
+    getPrecedingCharacter() {
+        var pos = this.selection.getCursor();
+
+        if (pos.column === 0) {
+            return pos.row === 0 ? "" : this.doc.getNewLineCharacter();
```

Comment:
What happens if indentation is automatically inserted after pressing newline? For example when I have

```
function() {
```

and I press enter after the last bracket, Ace would automatically inserting indentation but I would probably still like to get the auto-triggered suggestion.

Reply:
Great catch! I forgot to test that workflow after my second commit (https://github.com/ajaxorg/ace/pull/5519/commits/ae6b91aba9ef47e70cb536ea1ccc1e9cd578cda7). It wasn't triggering suggestions when indentation was automatically applied after a newline insertion.

Should be triggering correctly now with https://github.com/ajaxorg/ace/pull/5519/commits/a1ad3feb60f950281a595cd0bcc9674f6b2c4b72.